### PR TITLE
Article embeds

### PR DESCRIPTION
**Pull request by tomaskikutis**
TGA-66

requires https://github.com/superdesk/superdesk-core/pull/2500

https://github.com/superdesk/superdesk-client-core/assets/2076953/491f1c83-daa7-4ec8-a491-4ae5a0701194

### CODE REVIEW

File: scripts/core/editor3/directive.tsx [hunk 1]
```diff
@@ -245,6 +246,7 @@ class Editor3Directive {
                                     cleanPastedHtml={this.cleanPastedHtml}
                                     autocompleteSuggestions={autocompleteSuggestions}
                                     plainText={this.plainText}
+                                    canAddArticleEmbed={(srcId: string) => canAddArticleEmbed(srcId, this.item._id)}
```
Comment (thecalcc) [inline scripts/core/editor3/directive.tsx:249]: Doesn't type inference work here?

File: scripts/utils/dragging.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
+import {IArticle} from 'superdesk-api';
+
+/**
+ * NOTICE:
+ * if you want to test drag image in chrome, make sure to close dev tools first,
+ * otherwise this thumbnail image might not be displayed, depends on some performance settings there.
```
Comment (thecalcc) [inline scripts/utils/dragging.ts:6]: What do you mean by performance settings? Also, if you initially have dev tools open before rendering the thumbnail, when you render it, wouldn't it get resized accordingly?

File: scripts/core/editor3/components/article-embed/can-add-article-embed.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        ]).then(([srcProfile, destProfile]) => {
+            if (srcProfile.embeddable !== true) {
+                return {
+                    ok: false,
+                    error: gettext('Item content profile is not configured to allow embedding.'),
+                };
+            } else if (srcProfile.embeddable === true && destProfile.embeddable === true) {
+                return {
+                    ok: false,
+                    error: gettext('Can not embed to item which itself can be embedded.'),
```
Comment (thecalcc) [inline scripts/core/editor3/components/article-embed/can-add-article-embed.ts:35]: You can restructure `canAddArticleEmbedDefault` or at least extract the object into a constant to reuse it here.

Comment (tomaskikutis) [reply scripts/core/editor3/components/article-embed/can-add-article-embed.ts:35]: How do you suggest to reuse it? `canAddArticleEmbedDefault` is simply a fallback function when `_canAddArticleEmbed` is not passed.

File: scripts/core/editor3/directive.tsx (hunk 1 above)
Comment (tomaskikutis) [reply scripts/core/editor3/directive.tsx:249]: It doesn't. Because of multiple levels of wrapping.

File: scripts/utils/dragging.ts (hunk 2 above)
Comment (tomaskikutis) [reply scripts/utils/dragging.ts:6]: I've moved this file so comment was written probably a few years ago. I don't remember what are the performance settings.

> Also, if you initially have dev tools open before rendering the thumbnail, when you render it, wouldn't it get resized accordingly?

I don't get what you mean about resizing. That thumbnail is rendered in full size and is hidden, so there's never a reason to resize it. When you drag another element, you can set it as drag image. see https://codepen.io/webgeeker/full/KBzrxE/

File: scripts/core/editor3/components/article-embed/can-add-article-embed.ts (hunk 3 above)
Comment (thecalcc) [reply scripts/core/editor3/components/article-embed/can-add-article-embed.ts:35]: Ok, the messages are different in all cases, let's keep it as is.

**Review by thecalcc — APPROVED** [review]

**Comment by tomaskikutis** [comment]
Because I removed `authoring-react-post-broadcasting` branch, this PR got auto-closed and I couldn't change the base anymore. I've manually squashed-merged it to `authoring-refactoring-post-big-merge`

**Pull request closed without merging**